File: LeetCode/Easy/0344-reverse-string/0344-reverse-string.java
class Solution {
    public void reverseString(char[] s) {
        int left = 0, right = s.length - 1;

        while (left < right) {
            if (s[left] != s[right]) {
                char tmp = s[left];
                s[left] = s[right];
                s[right] = tmp;
            }

            left++;
            right--;
        }
    }
}

File: LeetCode/Easy/0344-reverse-string/0344-reverse-string.kt
class Solution {
    fun reverseString(s: CharArray): Unit {
        var left = 0
        var right = s.size - 1

        while (left < right) {
            if (s[left] != s[right]) {
                val tmp = s[left]
                s[left] = s[right]
                s[right] = tmp
            }

            left++
            right--
        }
    }
}